resources/osl_heisenbridge.rb

```diff
@@ -0,0 +1,38 @@
+# osl_heisenbridge
+#
+# Heisenbridge is an IRC bouncer for a Matrix server
+
+resource_name :osl_heisenbridge
+provides :osl_heisenbridge
+unified_mode true
+
+default_action :create
+
+property :container_name, String, name_property: true
+
+action :create do
+  # Pull down the heisenbridge image
+  docker_image 'hif1/heisenbridge'
+  
+  # Generate the app service file
+  execute 'heisenbridge app service' do
+    command "docker run --volume=/srv/synapse:/data hif1/heisenbridge  'python -m heisenbridge' -c /data/#{new_resource.container_name}.yaml --generate --listen-address #{new_resource.container_name}"
+    creates "/srv/synapse/#{new_resource.container_name}.yaml"
+  end
```

Comment:
Is there not another way of doing this? This is pretty hacky.

Reply:
This is Heisenbridge's [recommended way of generating an appservice file](https://github.com/hifi/heisenbridge/blob/master/docker-compose/init/heisenbridge-init.sh).

I was considering using a template instead, but decided to use this in the event Heisenbridge updates what is in the file, with this allowing us to easily adapt to the new content without needing to upgrade the cookbook.

Since Appservice files are usually [similar to each other](https://matrix-org.github.io/synapse/latest/application_services.html), the next build will instead be using a generic template instead.